SU2_CFD/src/solvers/CNEMONSSolver.cpp

```diff
@@ -804,99 +792,137 @@ void CNEMONSSolver::BC_IsothermalCatalytic_Wall(CGeometry *geometry,
+                                                                                 Cvtrs[iSpecies]  );
+            Jacobian_j[nSpecies+nDim][nSpecies+nDim+1] += Res_Visc[iSpecies]/Area*Cvve[iSpecies];
+          }
 
-        // Temperature
-        for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
-          for (jSpecies = 0; jSpecies < nSpecies; jSpecies++) {
-            Jacobian_j[nSpecies+nDim][iSpecies] += Jacobian_j[jSpecies][iSpecies]*hs[iSpecies];
+          // Vib.-El. Temperature
+          for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
+            for (jSpecies = 0; jSpecies < nSpecies; jSpecies++)
+              Jacobian_j[nSpecies+nDim+1][iSpecies] += Jacobian_j[jSpecies][iSpecies]*eves[iSpecies];
+            Jacobian_j[nSpecies+nDim+1][nSpecies+nDim+1] += Res_Visc[iSpecies]/Area*Cvve[iSpecies];
           }
-          Jacobian_j[nSpecies+nDim][nSpecies+nDim] += Res_Visc[iSpecies]/Area*(Ru/Ms[iSpecies] +
-                                                                               Cvtrs[iSpecies]  );
-          Jacobian_j[nSpecies+nDim][nSpecies+nDim+1] += Res_Visc[iSpecies]/Area*Cvve[iSpecies];
+
+          /*--- Multiply by the transformation matrix and store in Jac. ii ---*/
+          for (iVar = 0; iVar < nVar; iVar++)
+            for (jVar = 0; jVar < nVar; jVar++)
+              for (kVar = 0; kVar < nVar; kVar++)
+                Jacobian_i[iVar][jVar] += Jacobian_j[iVar][kVar]*dVdU[kVar][jVar]*Area;
+
+          /*--- Apply to the linear system ---*/
+          Jacobian.SubtractBlock(iPoint, iPoint, Jacobian_i);
         }
 
-        // Vib.-El. Temperature
-        for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
-          for (jSpecies = 0; jSpecies < nSpecies; jSpecies++)
-            Jacobian_j[nSpecies+nDim+1][iSpecies] += Jacobian_j[jSpecies][iSpecies]*eves[iSpecies];
-          Jacobian_j[nSpecies+nDim+1][nSpecies+nDim+1] += Res_Visc[iSpecies]/Area*Cvve[iSpecies];
+      } else {
+
+        if (implicit) {
+          SU2_MPI::Error("Implicit not currently available for partially catalytic wall.",CURRENT_FUNCTION);
         }
 
-        /*--- Multiply by the transformation matrix and store in Jac. ii ---*/
-        for (iVar = 0; iVar < nVar; iVar++)
-          for (jVar = 0; jVar < nVar; jVar++)
-            for (kVar = 0; kVar < nVar; kVar++)
-              Jacobian_i[iVar][jVar] += Jacobian_j[iVar][kVar]*dVdU[kVar][jVar]*Area;
+        /*--- Identify the boundary ---*/
+        string Marker_Tag = config->GetMarker_All_TagBound(val_marker);
+
+        /*--- Get isothermal wall temp ----*/
+        su2double Tw = config->GetIsothermal_Temperature(Marker_Tag);
+
+        /*--- Get wall catalytic efficiency ----*/
+        su2double gam = config->GetCatalytic_Efficiency();
+
+        /*--- Get gas model ---*/
+        string gas_model = config->GetGasModel();
+
+        if (gas_model == string("N2")) {
```

Comment:
string comparisons are not great for performance, I guess it's ok for boundary conditions but I would not rely in this pattern elsewhere.
I thought these models had enum values associated with them?

Reply:
The enum values do exist for the native solver, but we don't have them for the mutation++ connection.